### ISSUE TITLE
[calendar] Javascript date object always one day off

### ISSUE THREAD

**Issue by otavioserra** [fomantic/Fomantic-UI#2373]
# Bug Report

There are several crazy things that happen with a JS DATE object that convert strings. Please, put it problem on document of calendar module, because it happens with this module too: [https://stackoverflow.com/questions/7556591/is-the-javascript-date-object-always-one-day-off](https://stackoverflow.com/questions/7556591/is-the-javascript-date-object-always-one-day-off)

## Steps to reproduce
1. Using opt enabledDates and/or eventDates arrays with, for example: [ new Date('2022-06-02') ]

## Expected result
Correct date

## Actual result
Date's always one day off

## Version
2.8.8

**Comment by lubber-de** [comment]
See https://codeofmatt.com/javascript-date-parsing-changes-in-es6/
referenced by https://stackoverflow.com/a/29185654/6278591

TLDR; Try to use Slashes instead of dashes when using new Date() to have local time instead of UTC

The calendar module also supports raw strings instead of Date Objects, which might also help

**Comment by otavioserra** [comment]
I tried use slashes but the problem was the same. 

JSON source: "datas_disponiveis":{"2022-05-19":1,"2022-05-26":1,"2022-06-02":1}

I changed '-' for '/' in dates and the problem was solve using this:

var dateObj = new Date(data.replace(/-/g, '\/'));
